src/Diagnostics/Diagnostics.jl

```diff
@@ -0,0 +1,19 @@
+module Diagnostics
+
+export
+    run_diagnostic,
+    HorizontalAverage, Timeseries, FieldMaximum, CFL, AdvectiveCFL, DiffusiveCFL, NaNChecker
+
+using Oceananigans,
+      Oceananigans.Operators
+
+using Oceananigans: AbstractDiagnostic
+
+include("kernels.jl")
```

Comment:
may want to name this file `diagnostics_kernels.jl`. We should also probably rename the similarly named file in the nascent `TimeSteppers` submodule introduced in PR #505 `timestepping_kernels.jl`.

Reply:
I like the idea of more verbose file names. Done!